geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSScanIntegrationTest.java

```diff
@@ -54,295 +74,397 @@ public void tearDown() {
-    jedis.sadd("a", "1", "12", "3");
-
-    List<Object> result;
-
+    Set<byte[]> initialMemberData = initializeThousandMemberByteSet();
     List<byte[]> allEntries = new ArrayList<>();
-    String cursor = "0";
+    String cursor = ZERO_CURSOR;
 
     do {
-      result =
-          (List<Object>) jedis.sendCommand("a", Protocol.Command.SSCAN, "a", cursor, "COUNT", "2",
-              "COUNT", "1");
-      allEntries.addAll((List<byte[]>) result.get(1));
-      cursor = new String((byte[]) result.get(0));
-    } while (!Arrays.equals((byte[]) result.get(0), "0".getBytes()));
+      result = sendCustomSscanCommand(KEY, KEY, cursor, "COUNT", "1", "COUNT", "2");
+      cursor = result.getCursor();
+      allEntries.addAll(result.getResult());
+    } while (!cursor.equals(ZERO_CURSOR));
 
-    assertThat((byte[]) result.get(0)).isEqualTo("0".getBytes());
-    assertThat(allEntries).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes(),
-        "3".getBytes());
+    assertThat(allEntries).containsExactlyInAnyOrderElementsOf(initialMemberData);
   }
 
   @Test
-  public void givenMatch_returnsAllMatchingMembersWithoutDuplicates() {
-    jedis.sadd("a", "1", "12", "3");
-
+  public void givenSetWithThreeEntriesAndMatch_returnsOnlyMatchingElements() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
     ScanParams scanParams = new ScanParams();
     scanParams.match("1*");
 
-    ScanResult<byte[]> result =
-        jedis.sscan("a".getBytes(), "0".getBytes(), scanParams);
+    result = jedis.sscan(KEY.getBytes(), ZERO_CURSOR.getBytes(), scanParams);
 
     assertThat(result.isCompleteIteration()).isTrue();
-    assertThat(result.getResult()).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes());
+    assertThat(result.getResult()).containsOnly(FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes());
   }
 
   @Test
-  @SuppressWarnings("unchecked")
-  public void givenMultipleMatches_returnsMembersMatchingLastMatchParameter() {
-    jedis.sadd("a", "1", "12", "3");
+  public void givenSetWithThreeEntriesAndMultipleMatchArguments_returnsOnlyElementsMatchingLastMatchArgument() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
 
-    List<Object> result = (List<Object>) jedis.sendCommand("a", Protocol.Command.SSCAN, "a", "0",
-        "MATCH", "3*", "MATCH", "1*");
+    result = sendCustomSscanCommand(KEY, KEY, ZERO_CURSOR, "MATCH", "3*", "MATCH", "1*");
 
-    assertThat((byte[]) result.get(0)).isEqualTo("0".getBytes());
-    assertThat((List<byte[]>) result.get(1)).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes());
+    assertThat(result.getCursor()).isEqualTo(ZERO_CURSOR);
+    assertThat(result.getResult()).containsOnly(FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes());
   }
 
   @Test
-  public void givenMatchAndCount_returnsAllMembersWithoutDuplicates() {
-    jedis.sadd("a", "1", "12", "3");
-
+  public void givenSetWithThreeMembersAndMatchAndCount_returnsAllMatchingMembers() {
```

Comment:
Tests using both MATCH and COUNT are tricky. The way SSCAN works, first COUNT members are scanned, then the matching members (if any) from that list are returned, which means that it's entirely possible that 0 members are returned by a single SSCAN even if COUNT is not small and there are matching members in the set.

As such, it's difficult to know what to assert in a test for both COUNT and MATCH that's actually meaningful, as the number of members returned could be anything between 0 and every matching member (depending on hash collisions resulting in more than COUNT members being scanned in a single scan). I think the best bet would be to have a test that shows that a very large COUNT results in almost all of the matching members in the set being returned, to show that we didn't use the default value of COUNT, and that we filtered the scanned elements by MATCH:
```
  @Test
  public void givenLargeCountAndMatch_returnsOnlyMatchingMembers() {
    Set<byte[]> initialMemberData = initializeThousandMemberByteSet();

    ScanParams scanParams = new ScanParams();
    // There are 111 matching members in the set 0..999
    scanParams.match("9*");
    // Choose a large COUNT to ensure that some matching members are returned
    scanParams.count(950);
    result = jedis.sscan(KEY.getBytes(), ZERO_CURSOR.getBytes(), scanParams);

    List<byte[]> returnedMembers = this.result.getResult();
    // We know that we must have found at least 61 matching members, given the size of COUNT and the
    // number of matching members in the set
    assertThat(returnedMembers.size()).isGreaterThanOrEqualTo(61);
    assertThat(returnedMembers).isSubsetOf(initialMemberData);
    assertThat(returnedMembers).allSatisfy(bytes -> assertThat(new String(bytes)).startsWith("9"));
  }
```